### ISSUE TITLE
db migration fails durring update from 1.3.1 to 1.5.1

### ISSUE THREAD

**Issue by boontifex** [ComPlat/chemotion_ELN#1292]
- I am a system/ELN admin

DB Migration `bundle exec rails db:migrate` throws the following exception:

```
2023-04-21 04:27:46 +0000, [INFO]: Migrating to SampleSvgFileCorrection (20220816070825) 
== 20220816070825 SampleSvgFileCorrection: migrating ==========================
rake aborted!
StandardError: An error has occurred, this and all later migrations canceled:

no implicit conversion of nil into String
/chemotion/app/db/migrate/20220816070825_sample_svg_file_correction.rb:19:in `file?'
/chemotion/app/db/migrate/20220816070825_sample_svg_file_correction.rb:19:in `block in change'
```

I tried to figure out which records are affected using rails console: `bundle exec rails console`

```ruby
date = Time.new(2022, 6, 28)

molecules = Molecule.where('created_at > :date or updated_at > :date', date: date)
messy_m = []

molecules.find_each do |m|
  svg_file = m.send(:full_svg_path)
  unless svg_file && File.file?(svg_file)
    samples = Sample.where('molecule_id = :m_id', m_id: m.id)
    samples.find_each do |s|
      puts "User_Id: '#{s.user_id}', Molecule_Id: '#{s.molecule_id}', sample_svg_file: '#{s.sample_svg_file}'"
    end    
  end
end
```

So we figured out, that:

* one molecule has no svg_file
* 26 molecules svg files are missing in filesystem
* the affected molecules are related to samples without user_id (user_id is null)

So, whats the best way to deal with this situation?

**Comment by PiTrem** [comment]
until the next release is out , and if you are willing to tinker the db-migration files you can replace the following lines in 
db/migrate/20220816070825_sample_svg_file_correction.rb (line 19 and 31 ): 



```rb
 next unless File.file?(svg_file)
 ```
 with 
```rb
  next unless svg_file && File.file?(svg_file)
```


Also the following script would help rebuild SVG for those molecules (and samples) missing it.
(note that the Ketcher Service will return an SVG with no content if the molfile cannot be processed)


```rb
Molecule.find_each do |molecule|
  svg_path = molecule.send(:full_svg_path)
  next if svg_path && File.file?(svg_path)

  puts "Missing SVG for #{molecule.id}\n"

  if molecule.molfile.blank?
    puts "No molfile for #{molecule.id}\n"
    next
  end
  svg = Molecule.svg_reprocess('', molecule.molfile)

  if svg.blank?
    puts "cannot build SVG for #{molecule.id}\n"
    next
  end

  molecule.attach_svg(svg)
  molecule.save
  puts "SVG attached for #{molecule.id}\n"
end

Sample.find_each do |sample|
  svg_path = sample.send(:full_svg_path)
  next if svg_path && File.file?(svg_path)

  puts "Missing SVG for #{sample.id}\n"

  if sample.molfile.blank?
    puts "No molfile for #{sample.id}\n"
    next
  end
  svg = KetcherService::RenderSvg.svg(sample.molfile)

  if svg.blank?
    puts "Cannot build SVG for #{sample.id}\n"
    next
  end
  sample.sample_svg_file = svg
  sample.attach_svg
  sample.save
  puts "SVG attached for #{sample.id}\n"
end

```

**Comment by PiTrem** [comment]
This will be released in 1.5.4 tomorrow

Also with the new version, there is task to repocess the svg with ketcher-service (https://github.com/ComPlat/chemotion_ELN/pull/1308/commits/99f21bf9e846bb8fca49c0be59b017edb88dc7f3)

```sh
bundle exec rake 'svg:molecule[missing]'
```
